utils/utils.py

```diff
@@ -46,14 +46,16 @@ def _get_data_entities(token_claims: Dict,
         user_attributes:
 
     Returns:
-
+        List: The data entities list
     """
-    data_entities: List[Dict] = []
     if not token_claims:
-        return data_entities
+        return None
 
     # add roles from token
-    for role in token_claims['user_roles']:
+    groups = token_claims.get('user_roles') or token_claims.get('roles') or token_claims.get('groups')
```

Comment:
is it possible that groups is none? (if all 3 gets return none since none exists...) - if so, then you can do the get('...", []) so that you get an empty list in the worst case (its a list, right?)

Reply:
added another or with [] so the group in the worst case is an empty list.